01_fib/shahriys/main.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var outfib io.Writer = os.Stdout
+
+// Function to calculate nth Fibonacci or Negafibonacci number
+func nfib(n int) int64 {
```

Comment:
I like the separation of concerns and creating a recursive function to work out Fn. Nice work.

Reply:
While the separation of concern is usually a good thing, here it causes the performance hit, especially with this recursive implementation. Naive recursion runs in `O(2^n)` time, whereas iterative would be in `O(n)`. I would recommend going with the iterative solution at least. You can do it recursively in `O(n)` with tail recursion.
You can also print the results as you go if you don't want to have to recalculate each fibonacci number, but I like the separation of concerns as well, since this is more likely how it would look in practice.